tests/unit/data-table/DataTable.spec.js

```diff
@@ -0,0 +1,204 @@
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    { value: 'apple' },
+                    { value: 'orange' },
+                    { value: 'zebra' },
+                    { value: 'Zebras' },
+                    { value: 'Apples' }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'desc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual(['Zebras', 'zebra', 'orange', 'Apples', 'apple'])
+        })
+
+        it('Sorts a mix of strings and numbers naturally ascending', () => {
+            const localThis = {
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    { value: '1 dog' },
+                    { value: 12 },
+                    { value: '21 dogs' },
+                    { value: '3 dogs' },
+                    { value: 30 }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'asc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual(['1 dog', '3 dogs', 12, '21 dogs', 30])
+        })
+
+        it('Sorts a mix of strings and numbers naturally descending', () => {
+            const localThis = {
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    { value: 'device 1' },
+                    { value: 'device 3' },
+                    { value: 'device 21' }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'desc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual(['device 21', 'device 3', 'device 1'])
+        })
+
+        it('Treats null and undefined as equivalent to empty string', () => {
+            const localThis = {
+                lookupProperty: DataTable.methods.lookupProperty,
+                filterRows: DataTable.methods.filterRows,
+                rows: [
+                    {},
+                    { value: 'zebra' },
+                    { value: null },
+                    { value: 'apple' },
+                    { value: '' }
+                ],
+                sort: {
+                    key: 'value',
+                    order: 'asc'
+                }
+            }
+
+            expect(
+                DataTable.computed.filteredRows.call(localThis).map((row) => row.value)
+            ).toEqual([undefined, null, '', 'apple', 'zebra'])
```

Comment:
What is the result of these when sorting `desc`?
Should there be a test for that also?

Reply:
Added in 2b11e2c838eeafaab9fbc19146015f8ee1a8c88a